### PR TITLE
Migrate tests from "cmd/mattermost/commands/team_test.go" to use test…

### PR DESCRIPTION
**Pull request by akantsevoi**
Migrated server tests from using the t.Fatal method calls on their checks to use the testify toolkit in the file cmd/mattermost/commands/team_test.go

Fixes https://github.com/mattermost/mattermost-server/issues/12410

### CODE REVIEW

**Comment by akantsevoi** [comment]
@cpanato Sorry for disturbing, what does it mean "akantsevoi needs to sign the CLA" ? How can I fix that?

**Comment by akantsevoi** [comment]
@cpanato I fixed the 'akantsevoi needs to sign the CLA' issue. How can I trigger the rebuild action without fake push?

File: cmd/mattermost/commands/team_test.go [hunk 1]
```diff
@@ -67,14 +63,10 @@ func TestLeaveTeam(t *testing.T) {
...
-		t.Fatal("profile should not be on team")
-	}
+	require.False(t, found, "profile should not be on team")
 
 	if teams, err := th.App.Srv.Store.Team().GetTeamsByUserId(th.BasicUser.Id); err != nil {
-		if len(teams) > 0 {
-			t.Fatal("Shouldn't be in team")
-		}
+		require.Equal(t, 0, len(teams), "Shouldn't be in team")
 	}
```
Comment (saturninoabril) [inline cmd/mattermost/commands/team_test.go]: I suggest to have nil check on err as well, like
```go
teams, err := th.App.Srv.Store.Team().GetTeamsByUserId(th.BasicUser.Id)
require.Nil(t, err)
require.Equal(t, 0, len(teams), "Shouldn't be in team hello")
```

Comment (akantsevoi) [reply cmd/mattermost/commands/team_test.go]: agree, I'll make an update

**Comment by cpanato** [comment]
@akantsevoi looks like you already signed the CLA, thanks!

**Comment by DSchalla** [comment]
Thanks for your PR! :tada: Adding the check as @saturninoabril proposed is a good idea, otherwise looks good.

**Pull request merged**